src/services/codefixes/importFixes.ts

```diff
@@ -164,14 +167,14 @@ namespace ts.codefix {
         readonly moduleSpecifier: string;
         readonly importKind: ImportKind;
         readonly typeOnly: boolean;
+        readonly useRequire: boolean;
     }
 
     const enum ImportKind {
         Named,
         Default,
         Namespace,
-        Equals,
-        ConstEquals
+        CommonJS,

```

Comment:
This is the fundamental conceptual change in this file, and everything else is basically just implementation details. This enum used to represent final syntax; now it’s more a representation of where the symbol can be found in the module symbol. Whether or not to use JS is then calculated and stored elsewhere, so that

- `CommonJS` can mean `import foo = require('mod')` or `const foo = require('mod')`
- `Namespace` can mean `import * as foo from 'mod'` or `const foo = require('mod')`
- `Default` can mean `import foo from 'mod'` or `const { default: foo } = require('mod')`
- `Named` can mean `import { foo } from 'mod'` or `const { foo } = require('mod')`